anyway/views/news_flash/api.py

```diff
@@ -1,20 +1,53 @@
+# pylint: disable=no-name-in-module
+# pylint: disable=no-self-argument
+
 import datetime
 import json
 import logging
-from typing import List
 
-from flask import request, Response
-from sqlalchemy import and_, not_
+from typing import List, Optional
+
+from flask import request, Response, make_response, jsonify
+from sqlalchemy import and_, not_, or_
+
 
 from anyway.app_and_db import db
 from anyway.backend_constants import BE_CONST
 from anyway.models import NewsFlash
 from anyway.infographics_utils import is_news_flash_resolution_supported
 
+from pydantic import BaseModel, ValidationError, validator
+
 DEFAULT_OFFSET_REQ_PARAMETER = 0
 DEFAULT_LIMIT_REQ_PARAMETER = 100
 
 
+class NewsFlashQuery(BaseModel):
+
+    id: Optional[int]
+    road_number: Optional[int]
+    offset: Optional[int] = DEFAULT_OFFSET_REQ_PARAMETER
+    limit: Optional[int] = DEFAULT_LIMIT_REQ_PARAMETER
+    resolution: Optional[List[str]]
+    source: Optional[List[BE_CONST.Source]]
+    # Must set default value in order to allow access from "check_missing_date" validator when no value provided
+    # from the request
+    start_date: Optional[datetime.datetime] = None
+    end_date: Optional[datetime.datetime] = None
+
+    @validator("end_date", always=True)
+    def check_missing_date(cls, v, values):
+        if bool(v) != bool(values["start_date"]):
+            raise ValueError("Must provide both start and end date")
+        return v
+
+    @validator("resolution")
+    def check_supported_resolutions(cls, v):
+        supported_resolutions = get_supported_resolutions()
+        if not set(v) <= supported_resolutions:
+            raise ValueError(f"Resolution must be one of: {supported_resolutions}")
+        return v
```

Comment:
1. If we compare case insensitive, we should lower both values we compare.
2. It does not seem correct that we return `resolution` in Hebrew, but get the values to filter in English.